Sources/Shared/Protocols/MappableOnce.swift

```diff
@@ -0,0 +1,5 @@
+import Sugar
+
+public protocol MappableOnce {
+  init?(_ map: JSONDictionary) throws
```

Comment:
We could totally do that, but that would be a slightly bigger refactoring as both `relation` and `relations` create `Mappable` objects and those methods also need to handle the exceptions.
Not saying that it's a bad thing and we should totally look into it, however it would work for @kostiakoval needs without that :sunglasses: